small-talk/db/registerToDB.js

```diff
@@ -5,18 +5,29 @@ import bcrypt from "bcryptjs";
 export default async function handleRegister(username, password, firstname, lastname, dob){ // function that sends the credentials to the database
   const mongoboi = new Mongoboi(uri, "Users");
   const date = new Date(dob).getTime() / 1000; // return the inputted date as a Unix timestamp
-  await mongoboi.connect();
-
-  const hashedpass = await bcrypt.hash(password, 10);
-
+  //await mongoboi.connect();
+  
   const newUser = { // schema for holding the values into certain fields for database organization
     username: username,
-    password: hashedpass,
     firstname: firstname,
     lastname: lastname,
     dob: date,
   };
 
+  bcrypt.genSalt(10, async function(saltError, salt){ // generate a salt for password
+    if (saltError) {
+      return saltError
+    }
+    else {
+      bcrypt.hash(password, salt, function (hashError, hash){ // use hash algorithm to generate hashed password w/ salt
+        if (hashError){
+          return hashError
```

Comment:
In accountCreate's page, it's expecting null for an error in the account creation. If you're returning the error text for hashError or saltError, wouldn't it assume the account creation was successful?

Reply:
Would it be best to log the error here then return null?

Reply:
ah yes, this would be better to do! I am not too familiar with using bcrypt's syntax with using a function as a parameter, so I didn't really know what to return.

Reply:
would I log this using console.log()?

Reply:
I'm not sure what the structure of the hashError or the saltError are. It might be a string or an error object you need to fetch the string from. But yeah since we don't have a logging pipeline right now you can just use a console.log